zebra-network/src/peer/handshake.rs

```diff
@@ -705,9 +717,9 @@ where
         .await
         .ok_or(HandshakeError::ConnectionClosed)??;
     if let Message::Verack = remote_msg {
-        debug!("got verack from remote peer");
+        debug!("got verack message from remote peer");
     } else {
-        Err(HandshakeError::UnexpectedMessage(Box::new(remote_msg)))?;
+        debug!("ignoring non-verack message from remote peer");
```

Comment:
Blocker: This passes one specific test, but it isn't quite the behaviour we want.

We need to ignore any number of non-Verack messages, then return the successful Connection as soon as we get a Verack message.

(Currently, this code continues as soon as it gets any message - it doesn't wait for a Verack message.)

The code that waits for Version messages and the code that waits for Verack messages should look very similar.